### ISSUE TITLE
Implement Lion, up to 5x faster than Adam, and more accurate

### ISSUE THREAD

**Issue by PallHaraldsson** [FluxML/Flux.jl#2313]
### Motivation and description

https://arxiv.org/abs/2302.06675

>Lion  (EvoLved Sign Momentum). It is more memory-efficient than Adam as it only keeps track of the momentum. Different from adaptive optimizers, its update has the same magnitude for each parameter calculated through the sign operation. We compare Lion with widely used optimizers, such as Adam and Adafactor, for training a variety of models on different tasks. On image classification, Lion boosts the accuracy of ViT by up to 2% on ImageNet and saves up to 5x the pre-training compute on JFT.

It's 11 lines of pseudo-code (shorter than AdamW)

**Comment by ToucheSir** [comment]
Note that [subsequent research](https://arxiv.org/abs/2307.06440) has shown marginal at best improvements over Adam(W) with more rigorous experimental. Nevertheless, this should be a straightforward addition if anyone is interested in getting their feet wet with a PR.
